src/Main.ps1

```diff
@@ -899,6 +899,32 @@ function Invoke-Pester {
                 [PesterConfiguration] $PesterPreference = [PesterConfiguration]::Merge($callerPreference, $Configuration)
             }
 
+            if ($PesterPreference.Output.CIFormat.Value -eq 'Auto') {
+
+                # Variable is set to 'True' if the script is being run by a Azure Devops build task. https://docs.microsoft.com/en-us/azure/devops/pipelines/build/variables?view=azure-devops&tabs=yaml
+                # Do not fix this to check for boolean value, the value is set to literal string 'True'
+                if ($env:TF_BUILD -eq 'True') {
+                    $PesterPreference.Output.CIFormat = 'AzureDevops'
+                }
+                # Variable is set to 'True' if the script is being run by a Github Actions workflow. https://docs.github.com/en/actions/reference/environment-variables#default-environment-variables
+                # Do not fix this to check for boolean value, the value is set to literal string 'True'
+                elseif ($env:GITHUB_ACTIONS -eq 'True') {
+                    $PesterPreference.Output.CIFormat = 'GithubActions'
+                }
+
+                else {
+                    $PesterPreference.Output.CIFormat = 'None'
+                }
+            }
+
+            elseif ($PesterPreference.Output.CIFormat.Value -eq 'AzureDevops' -and $env:TF_BUILD -ne 'True') {
+                $PesterPreference.Output.CIFormat = 'None'
+            }
+
+            elseif ($PesterPreference.Output.CIFormat.Value -eq 'GithubActions' -and $env:GITHUB_ACTIONS -ne 'True') {
+                $PesterPreference.Output.CIFormat = 'None'
+            }
+
```

Comment:
The intention is good, but I would remove this. If the environment variables changes in the future, it's okay if auto-detect breaks, but explicit user settings should not break.

What do you think?

Reply:
Yeah I see your point. I think we probably just remove `AzureDevops` and `GithubActions` options and keep just `Auto` and `None`. In that case might it better to just use a boolean flag like `AutoDetectCI` as we discussed before?

Reply:
On the other hand keeping explicit CI options like this helps with determining what CI formatting to apply. I'd assume if we did `AutoDetectCI`, I would still need another property to set the CI that has been detected, since I currently pass this in into a hashtable with the CI header/message formatting.

Reply:
I've removed these lines and the associated tests.